File: Gateway/src/main/java/com/quizGpt/Gateway/Account/Entity/MqResponse.java
package com.quizGpt.Gateway.Account.Entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "RabbitMqResponse")
public class MqResponse {
    @Id
    private String correlationId;
    @NotNull
    @Column(length = Integer.MAX_VALUE)
    private String response;
}

File: resource-server/src/account/entities/mqResponse.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";

@Entity()
export class MqResponse {
  @PrimaryGeneratedColumn()
  correlationId: String;

  @Column()
  response: String;

  constructor(correlationId: String, response: String) {
    this.correlationId = correlationId;
    this.response = response;
  }
}
